### PR TITLE
feat(cmf): support simple name in settings props

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

Most of the time it's hard to know what name we can use in the `props` section.
Anwser it's the displayName of the component you cmfConnect.
So you can have `Container(ActionButton)#default` or `Translate(HeaderBar)#default`
This also means if we add an HOC we need to update props so it's a breaking change .

**What is the chosen solution to this problem?**

Let's remove HOC from the display name keeping compatibility.
So `ActionButton#default` and `HeaderBar#default` will match

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/cmf/src/settings.js [hunk 1]
```diff
@@ -21,6 +21,19 @@ export function generateDefaultViewId(viewId, componentName, componentId) {
 	return viewId;
 }
 
+/**
+ * this function remove all HOC from the componentName
+ * @param {String} viewId Connect(CMF(Container(MyComponent)))#default
+ * @return {String} MyComponent#default
+ */
+function withoutHOC(componentName) {
+	return componentName
```
Comment (jare-talend) [inline packages/cmf/src/settings.js]: It seems that you don't like the regular expression :)

File: packages/cmf/src/settings.js [hunk 2]
```diff
@@ -36,6 +49,11 @@ export function nonMemoized(state, ownProps, componentName, componentId) {
 
 	if (viewId && state.cmf.settings.props[viewId]) {
 		viewProps = state.cmf.settings.props[viewId] || {};
+	} else if (componentName && componentName.indexOf('(') !== -1) {
```
Comment (frassinier) [inline packages/cmf/src/settings.js]: I found that more readable:

`if (componentName && componentName.includes('(')) {`

File: packages/cmf/src/settings.js [hunk 3]
```diff
@@ -21,6 +21,19 @@ export function generateDefaultViewId(viewId, componentName, componentId) {
 	return viewId;
 }
 
+/**
+ * this function remove all HOC from the componentName
```
Comment (frassinier) [inline packages/cmf/src/settings.js]: removes

File: packages/cmf/src/settings.js (hunk 1 above)
Comment (talend-glorieux) [reply packages/cmf/src/settings.js]: No need for Regexp if they are not making more sense than simple string manipulation or are significantly faster.

Comment (jare-talend) [reply packages/cmf/src/settings.js]: split is lower than regexp. https://jsperf.com/split-regexp42
IMO, to get something in a string, I prefer use a regexp.

Comment (jmfrancois) [reply packages/cmf/src/settings.js]: done

**Pull request merged**